### ISSUE TITLE
Add CSV logging to LighterLogger

### ISSUE THREAD

**Issue by ibro45** [project-lighter/lighter#21]
## Description

Ideally, ~~match all the[ loggers that PyTorch Lightning provides](https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#supported-loggers).~~
- [x] TensorBoard
- [x] Wandb
- [ ] ~~Comet~~
- [ ] ~~MLFlow~~
- [ ] ~~Neptune~~
- [ ] CSV logging (@kbressem I remember you have something like this in your `trainlib`, suggestions are welcome)

**Comment by kbressem** [comment]
Yes, I log train loss on a step level and metric on an epoch level. This allows to create nice graphs, similar to tensorboard. Also, I save individual  metrics per file, but currently they are overwritten each epoch, this could be changed. So it is not really logging but more continuously saving results. Still very helpful for me.

CSV looks like this:  

| filename | class 1 | class 2 | class n| 
|-------| -------| -------|-------| 
| /path/to/dicom1 | 0.87123 | 0.74623 | 0.12341 |
| /path/to/dicom2 | 0.45425 | 0.84234 | 0.77412 |

I could imagine that logging every individual metric on an epoch level would also be helpful, to identify cases where the model has problems.

**Comment by ibro45** [comment]
@surajpaib and I discussed - we will not add support for other types of logging, we will have Tensorboard, W&B, and CSV logging support. Still need to implement the CSV one.

**Comment by ibro45** [comment]
https://github.com/project-lighter/lighter/pull/93 - LighterLogger removed, relying on PL loggers.

Should use https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.CSVLogger.html now. Haven't tried it ever. Have you guys?
